agent/src/com/jonnyzzz/teamcity/virtual/run/VMRunnerContext.java

```diff
@@ -68,22 +69,23 @@ public File getAgentTempDirectory() {
 
   @NotNull
   public String getCheckoutMountPoint() {
+    final String defaultMountPoint = SystemInfo.isUnix ? "/checkout" : "C:\\checkout";
     String mountPoint = myContext.getRunnerParameters().get(VMConstants.PARAMETER_CHECKOUT_MOUNT_POINT);
-    return StringUtil.isEmptyOrSpaces(mountPoint) ? "/checkout" : mountPoint;
+    return StringUtil.isEmptyOrSpaces(mountPoint) ? defaultMountPoint : mountPoint;
   }
 
   @NotNull
   public String getShellLocation() {
-    String loc = myContext.getRunnerParameters().get(VMConstants.PARAMETER_SHELL);
-    if (loc == null) {
+    final String loc = myContext.getRunnerParameters().get(VMConstants.PARAMETER_SHELL);
+    if (loc.equals("default")) {
       return "/bin/bash";
```

Comment:
I left here vagrant implementation. Docker's one lie in method `isDockerServerWindowsBased()`. Shall I merge them?